src/redux/auto-opponent/saga.ts

```diff
@@ -0,0 +1,52 @@
+import { put, take, actionChannel } from 'redux-saga/effects';
+import { delay } from 'redux-saga';
+
+import * as autoOpponentActions from './actions';
+
+import { fromProposal, GameEngine } from '../../game-engine/GameEngine';
```

Comment:
This is the only place that uses `fromProposal` now. Otherwise, `GameEngineB.fromProposal` is used directly. We should unify this.